plutus-core/plutus-core/stdlib/PlutusCore/StdLib/Type.hs

```diff
@@ -744,10 +744,12 @@ makeRecursiveType1
 makeRecursiveType1 ann dataName argVar patBody1 = do
     let varName = _tyVarDeclName argVar
         varKind = _tyVarDeclKind argVar
+    varName' <- freshenTyName varName
+    let
         recKind = KindArrow ann varKind $ Type ann
         pat1 = TyLam ann dataName recKind $ TyLam ann varName varKind patBody1
         -- recType = \(v :: k) -> ifix (\(dataName :: k -> *) (v :: k) -> patBody1) v
-        recType = TyLam ann varName varKind . TyIFix ann pat1 $ TyVar ann varName
+        recType = TyLam ann varName' varKind . TyIFix ann pat1 $ TyVar ann varName'
```

Comment:
Oh, that was really stupid on my part. I'm gonna look in this file to see if I made any other such stupid mistake. Thanks for fixing.